src/Idris/Parser.idr

```diff
@@ -1359,8 +1359,10 @@ dataOpt : OriginDesc -> Rule DataOpt
 dataOpt fname
     = (decorate fname Keyword (exactIdent "noHints") $> NoHints)
   <|> (decorate fname Keyword (exactIdent "uniqueSearch") $> UniqueSearch)
-  <|> (do decorate fname Keyword (exactIdent "search")
-          SearchBy <$> forget <$> some (decorate fname Bound name))
+  <|> (do b   <- bounds $ decorate fname Keyword (exactIdent "search")
+          det <- mustWorkBecause b.bounds "Expected list of determining parameters" $
```

Comment:
```suggestion
          det <- mustWorkBecause b.bounds "Expected at least one determining parameter" $
```

I think we can slightly improve the message, although I'm not thrilled with this change either, but due to the current error reporting framework we use I think that's the best we can do.

Reply:
I think a user is more likely to write something incorrect after `|` than leave the list blank. Therefore, I would leave the message more general.
```idris
Error: Expected list of determining parameters.

Test:1:17--1:18
 1 | interface Iface | "hello" where
                     ^
```

Reply:
Oh I see, well so be it. One day someone's going to have to overhaul all those messages anyways